api/v1alpha1/applicationset_types.go

```diff
@@ -181,6 +182,18 @@ type SCMProviderGeneratorGithub struct {
 	AllBranches bool `json:"allBranches,omitempty"`
 }
 
+// SCMProviderGeneratorGitlab defines a connection info specific to Gitlab.
+type SCMProviderGeneratorGitlab struct {
+	// Gitlab org to scan. Required.
+	Group string `json:"group"`
```

Comment:
Gitlab has the context of a group (which is sort of similar to a organization, except that a group can contain subgroups). This allows for certain kind of structures:

```
head-group
  subgroup-a
    repository-1
    repository-2
  repository-3
  subgroup-b
```

Github also has groups but they are more focused on users (teams). So it might be confusing to give both the same name

Reply:
Group is the term in gitlab, and I made this so it always includes subgroups, although now that I think about it, I should have made that an option in the gitlab spec.  I'll add that now.

The issue I was dealing with for organizations vs groups is that the scmprovider interface requires one of the methods to take a Repository, which includes an Organization field -- so I made the gitlab config take a group, but in the actual code it refers to "organization" to make it easier for me to track what's going on.

There may be a better way to handle that, but I was trying to avoid altering generic the scm provider code.

Reply:
I updated the comment for this one to make it more clear how this works.

Reply:
I also updated the docs

Reply:
Look great, thanks for the explanation :sunglasses: